File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    # 1. 수포자들의 패턴
    patterns = [
        [1, 2, 3, 4, 5],
        [2, 1, 2, 3, 2, 4, 2, 5],
        [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    ]
    
    # 2. 수포자 점수 저장 리스트
    scores = [0] * 3
    
    # 3. 각 수포자의 패턴과 정답이 얼마나 일치하는지 확인
    for i, answer in enumerate(answers):
        for j, pattern in enumerate(patterns):
            if answer == pattern[i % len(pattern)]:
                scores[j] += 1
    
    # 4. 가장 높은 점수 저장
    max_score = max(scores)
    
    # 5. 가장 높은 점수를 가진 수포자들의 번호를 찾아서 리스트에 담기
    answer = []
    for i, score in enumerate(scores):
        if score == max_score:
            answer.append(i+1)
    
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.ArrayList;

class Solution {
    public int[] solution(int[] answers) {
        int[] a = {1, 2, 3, 4, 5};
        int[] b = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] c = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int[] score = new int[4]; 
        
        int size = answers.length;
        
        for (int i = 0; i < size; i++){
            if (answers[i] == a[i%a.length]){
                score[1]++;
            }
            if (answers[i] == b[i%b.length]){
                score[2]++;
            }
            if (answers[i] == c[i%c.length]){
                score[3]++;
            }
        }
        
        int max = 0;
            
        for (int i = 1; i < score.length; i++){
            max = Math.max(score[i], max);
        }
        
        ArrayList<Integer> list = new ArrayList<>();
        
        for (int i = 1; i < score.length; i++){
            if (score[i] == max){
                list.add(i);
            }
        }
        int[] answer = new int[list.size()];
        for (int i = 0; i < list.size(); i++){
            answer[i] = list.get(i);
        }
        
        return answer;
    }
}